File: app/src/main/java/com/example/chap20todolist/FileHelper.java
package com.example.chap20todolist;

import android.content.Context;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.util.ArrayList;

public class FileHelper {

    public static final String FILENAME = "listInfo.dat";

    public static void writeData(ArrayList<String> item, Context context) {
        try {
            FileOutputStream fos = context.openFileOutput(FILENAME, Context.MODE_PRIVATE);
            ObjectOutputStream oos = new ObjectOutputStream(fos);
            oos.writeObject(item);
            oos.close();
        } catch (FileNotFoundException e) {
            throw new RuntimeException(e);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    public static ArrayList<String> readData(Context context) {
        ArrayList<String> item = null;
        try {
            FileInputStream fis = context.openFileInput(FILENAME);
            ObjectInputStream ois = new ObjectInputStream(fis);
            item = (ArrayList<String>) ois.readObject();
        } catch (FileNotFoundException e) {
            item = new ArrayList<>();
        } catch (IOException e) {
            throw new RuntimeException(e);
        } catch (ClassNotFoundException e) {
            throw new RuntimeException(e);
        }
        return item;
    }
}

File: app/src/main/java/com/example/chap10todolist/FileHelper.kt
package com.example.chap10todolist

import android.content.Context
import java.io.FileInputStream
import java.io.FileNotFoundException
import java.io.FileOutputStream
import java.io.ObjectInputStream
import java.io.ObjectOutputStream

class FileHelper {

    val FILENAME = "listInfo.dat"

    fun writeData(item: ArrayList<String>, context: Context) {
        var fos: FileOutputStream = context.openFileOutput(FILENAME, Context.MODE_PRIVATE)
        var oas = ObjectOutputStream(fos)

        oas.writeObject(item)
        oas.close()
    }

    fun readData(context: Context): ArrayList<String> {
        var itemList: ArrayList<String>

        try {
            var fis: FileInputStream = context.openFileInput(FILENAME)
            var ois = ObjectInputStream(fis)

            itemList = ois.readObject() as ArrayList<String>
        } catch (e: FileNotFoundException) {
            itemList = ArrayList()
            itemList.clear()
        }

        return itemList
    }

}